pkg/apis/flows/v1alpha1/flow_validation_test.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/knative/pkg/apis"
+	"k8s.io/api/core/v1"
+	"testing"
+)
+
+func TestFlowSpecValidation(t *testing.T) {
+	tests := []struct {
+		name string
+		f    *FlowSpec
+		want *apis.FieldError
+	}{{
+		name: "valid with URI",
+		f: &FlowSpec{
+			Action: FlowAction{
+				TargetURI: stringPtr("http://example.com/action"),
+			},
+			Trigger: EventTrigger{
```

Comment:
This gets recreated a few times. Maybe pull it into a variable? 

validEventTrigger := ...